WordPress/src/main/java/org/wordpress/android/ui/mlp/LayoutsAdapter.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.mlp
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.ImageView.ScaleType.FIT_CENTER
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.modal_layout_picker_layouts_card.view.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageType
+import org.wordpress.android.util.setVisible
+import javax.inject.Inject
+
+/**
+ * Renders the Layout cards
+ */
+class LayoutsAdapter(
+    private val context: Context,
+    private val selectionListener: LayoutSelectionListener
+) : RecyclerView.Adapter<LayoutsAdapter.ViewHolder>() {
+    @Inject lateinit var imageManager: ImageManager
+
+    private var layouts: List<LayoutListItem> = listOf()
+
+    init {
+        (context.applicationContext as WordPress).component().inject(this)
+    }
+
+    fun setData(data: List<LayoutListItem>) {
+        val diffResult = DiffUtil.calculateDiff(
+                LayoutsDiffCallback(
+                        layouts,
+                        data
+                )
+        )
+        layouts = data
+        diffResult.dispatchUpdatesTo(this)
+    }
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): ViewHolder {
+        val v = LayoutInflater.from(parent.context)
+                .inflate(R.layout.modal_layout_picker_layouts_card, parent, false)
+        return ViewHolder(v)
+    }
+
+    override fun getItemCount(): Int {
+        return layouts.size
+    }
+
+    override fun onBindViewHolder(
```

Comment:
We might want to consider extracting the viewHolder into a separate class to follow the SRP. Something like [this](https://github.com/wordpress-mobile/WordPress-Android/blob/b71ea8fe1ba4f4f3666881bfd6ad50066b9ef92f/WordPress/src/main/java/org/wordpress/android/ui/reader/discover/viewholders/ReaderPostViewHolder.kt#L31) just way simpler. Wdyt?

Reply:
I agree with you @malinajirka . Fixed with 5bf4fb52f048808ba7a73fe794037f055d1c5e90